packages/rum-core/src/browser/locationChangeObservable.ts

```diff
@@ -38,10 +38,10 @@ export function createLocationChangeObservable(location: Location) {
 }
 
 function trackHistory(onHistoryChange: () => void) {
-  const { stop: stopInstrumentingPushState } = instrumentMethodAndCallOriginal(history, 'pushState', {
+  const { stop: stopInstrumentingPushState } = instrumentMethodAndCallOriginal(History.prototype, 'pushState', {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Could we also have an issue for xhr.onreadystatechange:
https://github.com/DataDog/browser-sdk/blob/d5b4a4cb3ec0dfc44918cb6e6da031b8811fee5a/packages/core/src/browser/xhrObservable.ts#L89

Reply:
Ah yes, true. My approach does not work here, because:
* `onreadystatechange` is defined on the prototype, so we would need to call `instrumentMethodAndCallOriginal` with `XMLHttpRequest.prototype`
* but we need to set the `onreadystatechange` property on the instance, not the prototype, in order to be used as an event listener

Contrary to other methods like `window.open` where it could be set readonly without breaking what it does, setting an event listener property as readonly would break its usage, so in practice I don't think it will happen on purpose. Thus, I think it's fine to leave it like that. WDYT?

Reply:
> Contrary to other methods like window.open where it could be set readonly without breaking what it does, setting an event listener property as readonly would break its usage, so in practice I don't think it will happen on purpose.

LGTM